### PR TITLE
[Develop] Fave 17574 Time Picker

### PR DESCRIPTION
**Pull request by Kelvin-GohKengBoon**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots (if applicable)

Please add appropriate screenshots.

### CODE REVIEW

File: src/DatePicker/index.tsx [hunk 1]
```diff
@@ -1,33 +1,114 @@
...
+export const TimePicker: React.FC<TimePickerProps> = ({ onOpenChange ,className, ...restProps }) => {
+  const [open, setOpen] = React.useState(false);
+
+  const handleOpenChange = (isOpen:boolean) =>{
+    setOpen(isOpen);
+    if(onOpenChange){
+      onOpenChange(isOpen);
+    }
+  }
+  const style = classNames(className,open ? "ui-open" : "");
```
Comment (AladdinNullptr) [inline src/DatePicker/index.tsx:71]: Shared code can be moved outside the components and re-used. e.g: `handleOpenChange` can be moved outside  with a small change: `(isOpen: boolean, onOpenChange: (open: boolean)=>void)` and same for style

File: src/DatePicker/index.tsx [hunk 2]
```diff
@@ -1,33 +1,114 @@
...
+  return(
+    <RangePicker className={style} onOpenChange={handleOpenChange} picker="time" mode={undefined} ref={ref} {...restProps}/>
+  );
```
Comment (naimdanialjohar) [inline src/DatePicker/index.tsx:92]: why is the mode set as undefined? what does it do?

**Review by naimdanialjohar — COMMENTED** [review]
LGTM. let's wait for @AladdinNullptr to give his input

File: src/DatePicker/index.tsx [hunk 3]
```diff
@@ -1,33 +1,106 @@
...
+  return(
+    <BaseDatePicker className={getStyle(open,className)} onOpenChange={handleOpenChange} {...restProps}/>
+  );
+}
+
+export const QuarterPicker : React.FC<DatePickerProps> = ({ onOpenChange ,className, ...restProps }) =>{
+  const [open, setOpen] = React.useState(false);
+  const handleOpenChange = createHandleOpenChange(onOpenChange,setOpen);
+  return(
+    <BaseDatePicker className={getStyle(open,className)} onOpenChange={handleOpenChange} {...restProps} picker='quarter' format={"YYYY-'Q'Q"}/>
```
Comment (AladdinNullptr) [inline src/DatePicker/index.tsx:50]: To improve performance, you can wrap the style computation with `useMemo` and the open handler with `useCallback`

```suggestion
  const styles = React.useMemo(() => getStyle(open, className), [open, className])
  const handleOpenChange = React.useCallback(createHandleOpenChange(onOpenChange, setOpen), [onOpenChange, setOpen])

  return <BaseDatePicker className={styles} onOpenChange={handleOpenChange} {...restProps} />
```

File: src/index.tsx [hunk 4]
```diff
@@ -48,8 +48,10 @@ export {
   YearPicker,
   TimePicker,
   RangePicker,
-  QuarterPicker
-} from './DatePicker';
+  TimeRangePicker,
```
Comment (AladdinNullptr) [inline src/index.tsx:51]: missing export

**Review by AladdinNullptr — APPROVED** [review]

**Pull request merged**